synapse/events/spamcheck.py

```diff
@@ -208,9 +212,21 @@ async def check_registration_for_spam(
             # spam checker
             checker = getattr(spam_checker, "check_registration_for_spam", None)
             if checker:
-                behaviour = await maybe_awaitable(
-                    checker(email_threepid, username, request_info)
-                )
+                checker_args = inspect.getfullargspec(checker)
+                # Provide auth_provider_id if the function supports it
+                if "auth_provider_id" in checker_args.args:
+                    behaviour = await maybe_awaitable(
+                        checker(
+                            email_threepid,
+                            username,
+                            request_info,
+                            auth_provider_id,
+                        )
+                    )
+                else:
+                    behaviour = await maybe_awaitable(
+                        checker(email_threepid, username, request_info)
+                    )
```

Comment:
it might be neater to factor the `await maybe_awaitable(...)` out of the if condtion?